docs/source/task-launch.rst

```diff
@@ -84,67 +117,122 @@ We can submit tasks from other tasks.  This allows us to make decisions while
-object that connects to the scheduler.  There is a convenience function to do
-this for you so that you don't have to pass around connection information.
-However you must use this function ``worker_client`` as a context manager to
-ensure proper cleanup on the worker.
+object that connects to the scheduler. There are three options for this:
+
+1. ``dask.compute`` and ``dask.delayed``
+2. ``get_client`` with ``secede`` and ``rejoin``
+3. ``worker_client``
+
+We demonstrate the use of these functions below. But before we do, here's some
+design considerations for each function:
+
+* ``dask.compute``
+    * After jobs collected from cluster, computation continues on the
+      scheduling slot that submitted the jobs.
+    * If using Dask objects, will perform some graph merging to perform less
+      computation and communication. See "`The compute function`_" for more
+      detail.
+* ``get_client``
+    * Will continue to occupy a scheduling slot, unless ``secede`` and
+      ``rejoin`` are called (otherwise may deadlock the cluster!).
+    * The worker running the function will possibly transition to a different
+      scheduling slot (the worker that submitted the jobs is labeled as "maybe
+      long running").
+* ``worker_client``
+    * Intended to submit long running jobs (the worker is labeled as "long
+      running")
+    * Implemented as a context manager, which provides some ease of use.
+    * Will ``secede`` from the current worker, and likely transition to another
+      worker.
+    * Establishing a connection to the scheduler takes roughly 10–20ms. It's
+      wise for the submitted jobs to be several times longer than this.
+
+Transitioning to another worker involves moving all of the functions internal
+state to another machine. The time cost of this will not be insignificant if a
+large array is present in memory.
+
+For more details on transitioning to different workers, see "`Scheduling
+Policies`_" and "`Scheduling State`_".
+
+.. _Scheduling Policies: https://distributed.readthedocs.io/en/latest/scheduling-policies.html
+.. _Scheduling State: https://distributed.readthedocs.io/en/latest/scheduling-state.html
+.. _The compute function: https://dask.pydata.org/en/latest/scheduler-overview.html#the-compute-function
+
+``dask.compute``
+~~~~~~~~~~~~~~~~
+
+``dask.compute`` behaves as normal: it submits the functions to the graph,
+optimizes for less bandwidth/computation and gathers the results.
 
 .. code-block:: python
 
-   from distributed import worker_client
+    from distributed import Client
+    from dask import delayed, compute
 
-   def process_all(data):
-       with worker_client() as e:
-           elements = e.scatter(data)
-           futures = e.map(process_element, elements)
-           analysis = e.submit(aggregate, futures)
-           result = analysis.result()
-       return result
+    @delayed
+    def fib(n):
+        if n < 2:
+            return n
+        a, b = compute(fib(n-1), fib(n-2))
```

Comment:
I would split this out so that we can comment that the results of calling `fib` are delayed

```python
a = fib(n - 1)  # these calls are delayed 
b = fib(n - 2)
a, b = compute(a, b)  # execute both in parallel
```